### ISSUE TITLE
[BUG] Missing third-party requests after click action lead to an 500 Error Page

### ISSUE THREAD

**Issue by ferrange** [microsoft/playwright#33801]
**Version**
1.49.0

**Steps to reproduce**
1. Access to the client's website.
3. Perform a click action on a button.
4. Wait until all the the third-party requests triggered by the click action are fulfilled.

**Expected behaviour**
When the triggered the third-party requests are successfully fulfilled, the third-party content is reloaded on the same page we were on the begining of the start action.

**Actual behaviour**
The first 2 third-party requests work as expected (pebble and session for example) but from the third one never receives a response (userdata for example) and then we are redirected to a 500 Error Page. Check this [Image](https://github.com/user-attachments/assets/d2f7ca22-05e2-475b-891f-206cdfe574eb) for more information.

**Additional context**
We tried several things:
1. Intercept the requests to make sure the request data is the correct one 
2. Intercept the response to mock it --> same result
3. Repeat the exact same steps using WebdriverIO --> behaves as expected
4. Repeat the exact same steps using Cypress --> behaves as expected

**Environment**
`System:`
`OS: Windows 10 64 bits`
`CPU: AMD Ryzen 7 5800H with Radeon Graphics 3.20 GHz`
`Binaries:`
`Node: 21.2.0 and 18.20.4`
`npm: 10.7.0`
`npmPackages:`
`@playwright/test: 1.49.0`
`playwright: 1.49.0`

**Comment by mxschmitt** [comment]
We need a self-contained reproduction which we can run locally in order to look into this issue. Can you provide one?

**Comment by ferrange** [comment]
> We need a self-contained reproduction which we can run locally in order to look into this issue. Can you provide one?

I understand that but for privacy reasons I'm not allowed to share more information :(

**Comment by mxschmitt** [comment]
I'll close it for now as per above. Feel free to re-file if you have something you can share.

**Comment by ferrange** [comment]
> I'll close it for now as per above. Feel free to re-file if you have something you can share.

Yeah such a helpful reply. It's more than usual the code can not be shared and more than obvious you guys have a bug. However, as there is no code to reproduce (a simple click and wait for third-party requests to fulfill) the easiest is to close the issue ;)
Thanks for your help, it's been decisive in order to stop using your tool
